vendor/knative.dev/pkg/injection/sharedmain/main.go

```diff
@@ -245,3 +287,89 @@ func flush(logger *zap.SugaredLogger) {
 	logger.Sync()
 	metrics.FlushExporter()
 }
+
+// parseAndGetConfigOrDie parses the rest config flags and creates a client or
+// dies by calling log.Fatalf.
+func parseAndGetConfigOrDie() *rest.Config {
+	var (
+		masterURL = flag.String("master", "",
+			"The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+		kubeconfig = flag.String("kubeconfig", "",
+			"Path to a kubeconfig. Only required if out-of-cluster.")
+	)
+	flag.Parse()
+
+	cfg, err := GetConfig(*masterURL, *kubeconfig)
+	if err != nil {
+		log.Fatalf("Error building kubeconfig: %v", err)
+	}
+
+	return cfg
+}
+
+// memStatsOrDie sets up reporting on Go memory usage every 30 seconds or dies
+// by calling log.Fatalf.
+func memStatsOrDie(ctx context.Context) {
+	msp := metrics.NewMemStatsAll()
+	msp.Start(ctx, 30*time.Second)
+
+	if err := view.Register(msp.DefaultViews()...); err != nil {
+		log.Fatalf("Error exporting go memstats view: %v", err)
+	}
+}
+
+// setupLoggerOrDie sets up the logger using the config from the given context
+// and returns a logger and atomic level, or dies by calling log.Fatalf.
+func setupLoggerOrDie(ctx context.Context, component string) (*zap.SugaredLogger, zap.AtomicLevel) {
+	loggingConfig, err := GetLoggingConfig(ctx)
+	if err != nil {
+		log.Fatalf("Error reading/parsing logging configuration: %v", err)
+	}
+	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	ctx = logging.WithLogger(ctx, logger)
+
+	return logger, atomicLevel
+}
+
+// setupConfigMapWatchOrDie establishes a watch of the configmaps in the system
+// namespace that are labeled to be watched or dies by calling log.Fatalf.
+func setupConfigMapWatchOrDie(ctx context.Context, logger *zap.SugaredLogger) *configmap.InformedWatcher {
+	kc := kubeclient.Get(ctx)
+	// Create ConfigMaps watcher with optional label-based filter.
+	var cmLabelReqs []labels.Requirement
+	if cmLabel := system.ResourceLabel(); cmLabel != "" {
+		req, err := configmap.FilterConfigByLabelExists(cmLabel)
+		if err != nil {
+			logger.With(zap.Error(err)).Fatalf("Failed to generate requirement for label %q")
+		}
+		logger.Infof("Setting up ConfigMap watcher with label selector %q", req)
+		cmLabelReqs = append(cmLabelReqs, *req)
+	}
+	// TODO(mattmoor): This should itself take a context and be injection-based.
+	cmw := configmap.NewInformedWatcher(kc, system.Namespace(), cmLabelReqs...)
```

Comment:
join the two lines?

Reply:
@vagababov  I guess here is the actual PR for the "vendor'ed" change: https://github.com/knative/pkg/pull/1084/files#diff-7a2878cf5c6550229d8ac30e0d5e74e2R348-R351